frontend/src/component/events/EventLog/EventActions.tsx

```diff
@@ -0,0 +1,115 @@
+import { type FC, useState } from 'react';
+import {
+    IconButton,
+    ListItemText,
+    MenuItem,
+    MenuList,
+    Popover,
+    styled,
+    Tooltip,
+    Typography,
+} from '@mui/material';
+import FileDownload from '@mui/icons-material/FileDownload';
+import type { EventSchema } from '../../../openapi';
+
+const StyledActions = styled('div')(({ theme }) => ({
+    display: 'flex',
+    justifyContent: 'center',
+}));
+
+const StyledPopover = styled(Popover)(({ theme }) => ({
+    borderRadius: theme.shape.borderRadiusLarge,
+    padding: theme.spacing(1, 1.5),
+}));
+
+interface IEventActions {
+    events: EventSchema[];
+}
+
+export const EventActions: FC<IEventActions> = ({ events }) => {
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+    const open = Boolean(anchorEl);
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setAnchorEl(event.currentTarget);
+    };
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+
+    const exportJson = () => {
+        const jsonString = JSON.stringify(events);
+        const blob = new Blob([jsonString], { type: 'application/json' });
+        const url = URL.createObjectURL(blob);
+
+        const currentDate = new Date().toISOString().split('T')[0];
+        const fileName = `events_${currentDate}.json`;
+
+        const a = document.createElement('a');
+        a.href = url;
+        a.download = fileName;
+        a.click();
+
+        URL.revokeObjectURL(url);
+        setAnchorEl(null);
+    };
+
+    const exportCsv = () => {
+        setAnchorEl(null);
+    };
+
+    return (
+        <StyledActions
+            onClick={(e) => {
+                e.preventDefault();
+                e.stopPropagation();
+            }}
+        >
+            <Tooltip title={'Export'} arrow describeChild>
+                <div>
+                    <IconButton
+                        aria-haspopup='true'
+                        aria-expanded={open ? 'true' : undefined}
+                        onClick={handleClick}
+                        type='button'
+                    >
+                        <FileDownload />
+                    </IconButton>
```

Comment:
I think this button needs some sort of text or label so that it's obvious what it does to non-sighted users.

If we don't want any visible text on the button, you can use the `ScreenReaderOnly` component, which'll make the contents visible to screen readers, but not rendered visibly on the screen.

I'm also not familiar with all the aria-stuff you need to make this accessible, but it seems like you've done a bit of research. Are the aria attributes based on something?

Also, according to [mdn on the aria-expanded attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded#values), `undefined` indicates that "the element does not own or control a grouping element that is expandable.", while `false` indicates that "the grouping element this element owns or controls is collapsed". So in this case, I think it should just be `aria-expanded={open}`

Reply:
I think the wrapper has tooltip text that will be read with screenreader.

Reply:
Or not..

Reply:
Added aria label.